### PR TITLE
Replaced deprecated StubJob with Mockito mock

### PR DESCRIPTION
**Pull request by StefanSpieker**
Replaced deprecated StubJob with Mockito mock. And since one test was just for testing the mocking class, I replaced this basic test with another test.
Alos some minor enhancements like removing superflous `throws Exception`

### Proposed changelog entries

* N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/test/java/hudson/model/JobTest.java [hunk 1]
```diff
@@ -26,24 +27,21 @@
...
        // call setDisplayNameFromRequest
        j.setDisplayNameOrNull(displayName);
        
        // make sure the displayname has been set
        assertEquals(displayName, j.getDisplayName());
     }
-
+    
     @Test
-    public void testSetDisplayNameZeroLength() throws Exception {
```
Comment (stellargo) [inline core/src/test/java/hudson/model/JobTest.java:39]: It looks like this test was removed and a different test was added. Might make sense to document reason for the same

**Review by stellargo — CHANGES_REQUESTED** [review]

**Review by MRamonLeon — APPROVED** [review]

File: core/src/test/java/hudson/model/JobTest.java (hunk 1 above)
Comment (res0nance) [reply core/src/test/java/hudson/model/JobTest.java:39]: @StefanSpieker could you address this bit?

**Review by oleg-nenashev — COMMENTED** [review]
TBH I would rather recommend using JenkinsRule instead of extending use of Mockito in integration tests, it becomes a burden to maintain them when doing complex API changes inside classes. I do not mind merging this pull request once the removed test comment is addressed.

**Comment by jglick** [comment]
I would also not advise investing effort into the likes of `core/src/test/java/hudson/model/JobTest.java`. If its functionality is not already well covered in `test/src/test/java/`, maybe better to just delete it. Or if you actually want unit tests, then do not use mocks—factor out the true logic into a package-accessible `static` method.

**Comment by basil** [comment]
Marking with `proposed-for-close`, as the maintainers have indicated that we would like to go in a different direction from the one proposed in this PR.

**Pull request closed without merging**